hydra-main/src/main/java/com/addthis/hydra/query/aggregate/QueryTask.java

```diff
@@ -145,7 +182,74 @@ private static AutoField getSourceField(Query query) {
                 throw new UncheckedIOException(e);
             }
         } else {
-            return new NoopField();
+            return null;
```

Comment:
not sure why this was changed since it seems unrelated, but please try to use `@Nullable` annotations where applicable (or newly applicable).